presto-spi/src/main/java/com/facebook/presto/spi/function/SqlInvokedFunction.java

```diff
@@ -98,10 +121,19 @@ public SqlInvokedFunction(
                 .map(Parameter::getType)
                 .collect(collectingAndThen(toList(), Collections::unmodifiableList));
 
-        this.signature = new Signature(functionName, SCALAR, typeVariableConstraints, emptyList(), returnType, argumentTypes, false);
+        this.signature = new Signature(functionName, kind, typeVariableConstraints, emptyList(), returnType, argumentTypes, false);
         this.functionId = new SqlFunctionId(functionName, argumentTypes);
         this.functionVersion = requireNonNull(version, "version is null");
         this.functionHandle = version.hasVersion() ? Optional.of(new SqlFunctionHandle(this.functionId, version.toString())) : Optional.empty();
+
```

Comment:
we should requireNonNull for aggregationMetadata.  That's the whole point of using an Optional.

Reply:
Will update. Thanks.